File: app/src/main/java/at/tomtasche/reader/background/ParcelUtil.java
package at.tomtasche.reader.background;

import android.os.Parcel;

public class ParcelUtil {

    public static void writeBoolean(Parcel parcel, boolean value) {
        parcel.writeInt(value ? 1 : 0);
    }

    public static boolean readBoolean(Parcel parcel) {
        return parcel.readInt() != 0;
    }
}

File: app/src/main/java/at/tomtasche/reader/background/ParcelUtil.kt
package at.tomtasche.reader.background

import android.os.Parcel

/** Parcel.writeBoolean/readBoolean exist since API 29, which is above minSdk. */
object ParcelUtil {

    @JvmStatic
    fun writeBoolean(parcel: Parcel, value: Boolean) {
        parcel.writeInt(if (value) 1 else 0)
    }

    @JvmStatic fun readBoolean(parcel: Parcel): Boolean = parcel.readInt() != 0
}
